met-web/src/services/widgetService/index.tsx

```diff
@@ -0,0 +1,44 @@
+import http from 'apiManager/httpRequestHandler';
+import Endpoints from 'apiManager/endpoints';
+import { replaceUrl } from 'helper';
+import { Widget, WidgetsList } from 'models/widget';
+
+export const getWidgets = async (engagement_id: number): Promise<WidgetsList[]> => {
+    const url = replaceUrl(Endpoints.Widgets.GET_LIST, 'engagement_id', String(engagement_id));
+    const responseData = await http.GetRequest<WidgetsList[]>(url);
+    return responseData.data.result ?? [];
+};
+
+interface PostWidget {
+    widget_type_id: number;
+    engagement_id: number;
+}
+export const postWidget = async (engagement_id: number, data: PostWidget): Promise<Widget[]> => {
```

Comment:
do we need the interface here? we already get the engagemend_id ..the only param we need here is widget_type_id which can be just a param..
But i think in future , we can accomdate more things this way..so ok either way

Reply:
Noted, I just felt to separate the body and the param for the reason you mentioned above and just to allow perhaps removing the param if we didn't like it and we don't have to change any other logic.